src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestOpArgument.ll

```diff
@@ -1,4 +1,4 @@
-define %String* @Microsoft__Quantum__Testing__QIR__TestOpArgument__body() #0 {
+define %String* @Microsoft__Quantum__Testing__QIR__TestOpArgument__body() {
```

Comment:
This IR seems like it went from being an entrypoint to no longer being an entrypoint, but I don't see a corresponding change in any qs files to remove that attribute. Is there something I'm misunderstanding here?

Reply:
There is still an entry point in the file, just a separate one. The QIR generation emits the "normal" QIR compilant definition of the entry point function. Additionally, it generates a function that calls into that entry point and has a signature that does not rely on any of the runtime types, or contain any of the opaque QIR types and uses 8-bit multiples for intergers and similar things. If the signatures of both functions match, then the wrapper is not generated but the QIR function is just tagged as entry point (which was previously the case for TestOpArgument, but is no longer the case since we started to properly map strings).

Reply:
Thanks, that clarifies it!